dagshub/data_engine/model/datasource.py

```diff
@@ -57,7 +57,10 @@
 from dagshub.data_engine.model.query import QueryFilterTree
 from dagshub.data_engine.model.schema_util import metadataTypeLookup, metadataTypeLookupReverse
 from dagshub.data_engine.model.datasource_state import DatasourceState
-
+# 👇 Required for running adding of LS backend from a notebook
+import asyncio
+import nest_asyncio
+nest_asyncio.apply()
```

Comment:
1) This new package hasn't been added to the requirements
2) The package is archived as of May this year: https://github.com/erdewit/nest_asyncio
I am not super comfortable adding a dead package like that, considering that it's doing things to the python internals that are possible to change on a new Python release. Is there another way to do this? What happens if you spin off a regular thread?

Reply:
Also just as a general note, putting this at the top of the file means it will be applied to EVERYONE who ever uses the client.
This is a pretty dangerous and intrusive change, so if we do need to resort to this in the end, it should be localized to the ngrok forwarding function only.